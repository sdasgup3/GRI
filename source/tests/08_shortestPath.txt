function displayAdjacencyMatrix(matrix, vertices)
{
  print("   ");
  foreach(vertex; vertices.iterator()) {
    print("" + vertex.__id + " ");
  }

    println("");
  it = vertices.iterator();
  foreach(line; matrix.iterator()) {
    if(it.hasNext())
      vertex = it.next();

    print("" + vertex.__id + " |");
        
    foreach(item; line.iterator())  {
      print("" + item + " ");
    }

    println("");
  }
}

function displaySTP(pathNdist, start, end, numvertices)
{

  parent = pathNdist[0];
  dist = pathNdist[1];

  //println("Print distance / path");

  if(-1 != end) {
    path = getPath(parent, start, end);

    print("" + end + " | " + dist[end] + " | ");
    for(i = 0 ; i < path.size(); i++) {
      print("" + path[i] + " ");
    }
    println("");

  } else {
    for(i = 0 ; i < numvertices ; i++) {
      end = i;
      //path = getPath(parent, start, end);

      print("" + end + " | " + dist[i] + " | " + parent[i]);
      //for(j = 0 ; j < path.size() -1; j++) {
     //   print("" + path[j] + " ");
     // }
      println("");
    }
  }
}



function getPath(parent, start, end)
{
  path = array(1);
  runner = end;
  while(runner != start) {
    path.pushFront(runner);
    runner = parent[runner];
  }
  path.pushFront(start);

  return path;
}

function main(argv) 
{
  g = graph();
  //g.loadFromFile("graphs/simple_shortestpath.txt");
  g.loadFromFile(argv[0]);
  //g.loadFromFile("graphs/02_graph_5_5_5.txt");

  vertices = g.getVertices();
  it = vertices.iterator();
  start = it.next();
  //it.next();
  //it.next();
  //it.next();
  //end = it.next();

  //matrix = g.getAdjacencyMatrix();
  //displayAdjacencyMatrix(matrix, vertices);


  pathNdist = g.getShortestPath("__w", start, NULL);
  displaySTP(pathNdist, start.__id, -1, g.getNumVertices());

  //pathNdist = g.getShortestPath("__w", start, end);
  //displaySTP(pathNdist, start.__id, end.__id, g.getNumVertices());

}



