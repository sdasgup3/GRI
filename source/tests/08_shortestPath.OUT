   0 1 2 3 4 5 6 7 8 
0 |0 1 0 0 0 0 0 1 0 
1 |1 0 1 0 0 0 0 1 0 
2 |0 1 0 1 0 1 0 0 1 
3 |0 0 1 0 1 1 0 0 0 
4 |0 0 0 1 0 1 0 0 0 
5 |0 0 1 1 1 0 1 0 0 
6 |0 0 0 0 0 1 0 1 1 
7 |1 1 0 0 0 0 1 0 1 
8 |0 0 1 0 0 0 1 1 0 
Print distance / path
0 | 0 | 0 NULL 
1 | 4 | 0 1 NULL 
2 | 12 | 0 1 2 NULL 
3 | 19 | 0 1 2 3 NULL 
4 | 21 | 0 7 6 5 4 NULL 
5 | 11 | 0 7 6 5 NULL 
6 | 9 | 0 7 6 NULL 
7 | 8 | 0 7 NULL 
8 | 14 | 0 1 2 8 NULL 
Print distance / path
4 | 21 | 0 7 6 5 4 NULL 
